hedera-node/src/main/java/com/hedera/services/txns/contract/ContractCallTransitionLogic.java

```diff
@@ -135,40 +137,53 @@ public void doStateTransition() {
 							.setRealmNum(op.getContractID().getRealmNum())
 							.setShardNum(op.getContractID().getShardNum()).build();
 			Address receiver = Address.fromHexString(asSolidityAddressHex(receiverAccount));
-			Wei gasPrice = Wei.of(1);
-			long gasLimit = op.getGas();
-			if (gasLimit > properties.maxGas()) {
-				gasLimit = properties.maxGas();
-			}
-
-			Wei value = Wei.ZERO;
-			if (op.getAmount() > 0) {
-				value = Wei.of(op.getAmount());
-			}
-
-			var evmTx = new Transaction(0, gasPrice, gasLimit, Optional.of(receiver), value, null, Bytes.fromHexString(CommonUtils.hex(op.getFunctionParameters().toByteArray())), sender, Optional.empty());
-			var defaultMutableWorld = new DefaultMutableWorldState(this.store);
-			var updater = defaultMutableWorld.updater();
-			var result = txProcessor.processTransaction(
-					stubbedBlockchain(),
-					updater,
-					stubbedBlockHeader(txnCtx.consensusTime().getEpochSecond()),
-					evmTx,
-					Address.fromHexString(asSolidityAddressHex(txnCtx.submittingNodeAccount())),
-					OperationTracer.NO_TRACING,
-					false);
-			var contractFunctionResult = ContractFunctionResult.newBuilder()
-					.setGasUsed(result.getEstimateGasUsedByTransaction())
-					.setErrorMessage(result.getRevertReason().toString());
-			Optional.ofNullable(result.getOutput().toArray())
-					.map(ByteString::copyFrom)
-					.ifPresent(contractFunctionResult::setContractCallResult);
-			txnCtx.setCallResult(contractFunctionResult.build());
-			if (result.isSuccessful()) {
+			if (this.store.get(receiver) != null) {
```

Comment:
Instead of having `if-else`, wouldn't it be better to check whether it exists, if not -> set status and return. This way you would need to have a nested logic in it.
Plus, because you are referring to a given contract, the proper status should be `INVALID_CONTRACT_ID`.